runtime/gc_base/modronapi.cpp

```diff
@@ -878,10 +880,25 @@ j9gc_set_allocation_sampling_interval(J9JavaVM *vm, UDATA samplingInterval)
 {
 	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(vm);
 	if (0 == samplingInterval) {
-		/* avoid (env->_oolTraceAllocationBytes) % 0 which could be undefined. */
+		/* avoid (env->_traceAllocationBytes) % 0 which could be undefined. */
 		samplingInterval = 1;
 	}
 	extensions->oolObjectSamplingBytesGranularity = samplingInterval;
+	if ((UDATA_MAX != samplingInterval) && !extensions->disableInlineAllocationForSamplingBytesGranularity) {
+		extensions->disableInlineAllocationForSamplingBytesGranularity = true;
+
+		GC_VMThreadListIterator threadIterator(vm);
+		J9VMThread *walkThread = NULL;
+
+		while((walkThread = threadIterator.nextVMThread()) != NULL) {
+			MM_EnvironmentBase *env = MM_EnvironmentBase::getEnvironment(walkThread->omrVMThread);
+			if ((NULL != env) && (NULL != env->_objectAllocationInterface)) {
+				env->_objectAllocationInterface->disableCachedAllocations(env);
+			}
+		}
```

Comment:
This looks very scary... I am not telling it is wrong solution, but:
- can we do it inside allocation path for each thread? (like thread needs allocation, check `extensions->disableInlineAllocationForSamplingBytesGranularity` is set so call `env->_objectAllocationInterface->disableCachedAllocations(env)`). We are interested in threads doing actual allocation. Did I missed something?
- in case it must be done by external walk of all threads, should it be guarded by VM Access or even Exclusive?

@amicic What do you think?

Reply:
@dmitripivkine 
1, currently we check extensions->disableInlineAllocationForSamplingBytesGranularity only in case during out of line allocation, check it at inline case might be too expensive.
2, you are right, we might need VM Access or Exclusive VM Access for  external walk.

Reply:
Do you mean if we do it at allocation thread it is possible that thread would use current active TLH until it reach refresh point eventually where inline allocation can be disabled?